core/camel-core/src/test/java/org/apache/camel/processor/SagaTimeoutTest.java

```diff
@@ -64,6 +70,29 @@ public void testTimeoutHasNoEffectIfCompleted() throws Exception {
         compensate.assertIsNotSatisfied();
     }
 
+    @Test
+    public void testTimeoutMultiParticipants() throws Exception {
+
+        MockEndpoint compensate = getMockEndpoint("mock:compensate");
+        compensate.expectedMessageCount(1);
+
+        MockEndpoint complete = getMockEndpoint("mock:complete");
+        complete.expectedMessageCount(2);
```

Comment:
That should be `0` since the Saga participants are compensated none can be completed. The participants cannot be both completed and compensated.

Reply:
Oh, I missed that, it would be definitively more clear 😄 It makes me wonder in which case `assertIsNotSatisfied` can be useful, it sounds like a very dangerous method.